lib/cachex/warmer.ex

```diff
@@ -99,11 +102,11 @@ defmodule Cachex.Warmer do
 
           # set pairs without options
           {:ok, pairs} ->
-            Cachex.put_many(cache, pairs)
+            Cachex.put_many(name, pairs)
```

Comment:
This change was included because there was a race condition caused by the use of the `cache` here - we had perviously discussed that the performance impacts of this change were mostly likely irrelevant due the frequency of warmers - happy to try and address this in some other way if you have thoughs

> 2. Warmers should use the latest version of cache state.
>    
>    * I definitely consider this one a bug!
>    * There's no good reason to not use the name
>    * The overhead is so small, it will only ever cause negative side effects (like this PR) by using the state

Reply:
Yeah, totally spaced on this one - the bug also sort of exists on L117, to a lesser extent. I think the best approach here is to wrap everything in the `case` body in a `Cachex.execute(name)` block, so that we have a copy of the latest state regardless.

That also means that we can purely store the warmer's `state` and the timer ref as the server state, with just the cache name:

```
{ name, state, timer }
```

Reply:
Note to self that this kinda sucks because `provisions` don't work with warmers, only hooks. I don't think it should be changed in this PR, but I have filed https://github.com/whitfin/cachex/issues/332 to follow up on to clean this type of thing up a bit more.